### PR TITLE
Redux 2 - Activity Create Tested Working

### PR DESCRIPTION
**Pull request by plasticviking**
# Overview

This PR includes the following proposed change(s):

- {List all the changes, if possible add the linked issue/ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots if applicable.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/state/reducers/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import { createConfigurationReducerWithDefaultState } from './configuration';
+import { createUserInfoReducer } from './userInfo';
+import { createNetworkReducer } from './network';
+
+function createRootReducer(config: AppConfig) {
+  return combineReducers({
+    Configuration: createConfigurationReducerWithDefaultState(config),
+    Auth: createAuthReducer(config),
+    UserInfo: createUserInfoReducer({ loaded: true, accessRequested: false, activated: true }),
```
Comment (sam-warren) [inline app/src/state/reducers/index.ts]: Should this be initalized to activated: false? Activated determines whether or not a user's account has been activated by an admin and determines what tabs they can see when they first hit the site. By setting this as default true we're showing all the tabs to anyone who hits the site. Same goes for loaded: should probably be default false and set to true once the extendedInfo is ready.

File: app/src/components/tabs/TabsContainer.tsx [hunk 2]
```diff
@@ -129,46 +131,54 @@ export interface ITabConfig {
...
   };
 
-  // loadUserFromCache();
-  /*
-    Function to logout current user by wiping their keycloak access token
-  */
   const logoutUser = async () => {
     history.push('/home/landing');
-    await authContext.logoutUser();
+    dispatch({ type: 'AUTH_SIGNOUT_REQUEST' });
```
Comment (sam-warren) [inline app/src/components/tabs/TabsContainer.tsx]: This doesn't appear to be wired up to anything.

**Review by sam-warren — CHANGES_REQUESTED** [review]
Looks good but uncovered a few bugs in testing:

- Logout not working / not wired up
- Userinfo in TabsContainer is not wired up properly - unauthenticated user can see all tabs. Likely due to the initialization of the user info object in the store.
- Errors in requests to paths that should be public
- "Record Category" when creating a new record fails to fetch dropdown items
- "Plan My Trip" is visible on web when it should not be, feature gate not working
- There is currently no way to grab IDIR userid and BCEID userid. Need to either include those in extended info OR refactor to remove all occurrences of them if they are redundant.

File: app/src/state/sagas/auth.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  const authHeaders = yield select(selectAuthHeaders);
+
+  try {
+    const { data: userData } = yield Http.request({
+      method: 'GET',
+      url: configuration.API_BASE + `/api/user-access`,
+      headers: {
+        Authorization: authHeaders.authorization,
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Origin': '*'
```
Comment (sam-warren) [inline app/src/state/sagas/auth.ts]: Is this safe?

**Review by micheal-w-wells — APPROVED** [review]

**Comment by micheal-w-wells** [comment]
Great stuff - happy to see some state management refactoring... let's shrink changeset size and go for more incremental moving forward so we can catch branches vulnerable to git gotchas (like reverted PR's changing hashes causing conflicts, etc), and also make them easier to review, merge, and catch breaks before they get too big.

**Pull request merged**